src/utils/pxls/template_manager.py

```diff
@@ -826,3 +825,34 @@ def fast_max_chunk(chunked_mask):
             chunk_pixels = sum
             max_index = i
     return max_index
+
+
+def layer(
+    templates: Iterable[Template],
+    placemap: Optional[np.ndarray] = None,
+    crop_to_placemap=True,
+    crop_to_template=True,
+) -> np.ndarray:
+    """
+    Sequentially layer each of the received templates, and return the
+    corresponding palettized image. Result is cropped to the placemap.
+    """
+    if placemap is None:
+        placemap = stats.placemap_array
+    background = np.full_like(placemap, 255, dtype=np.uint8)
+    max_x, max_y = 0, 0
+    min_x, min_y = background.shape[1], background.shape[0]
+    for template in templates:
+        arr, ox, oy = template.crop_to_canvas()
+        # Don't update transparent pixels
+        mask = arr != 255
+        background[oy : oy + arr.shape[0], ox : ox + arr.shape[1]][mask] = arr[mask]
+        min_x = min(ox, min_x)
+        min_y = min(oy, min_y)
+        max_x = max(ox + template.width, max_x)
+        max_y = max(oy + template.height, max_y)
+    if crop_to_placemap:
+        background[placemap != 0] = 255
+    if crop_to_template:
+        return background[min_y:max_y, min_x:max_x]
+    return background
```

Comment:
This function throws an error if the template has some part outside the canvas (on the right or bottom) or if it's completely outside the canvas. That's why I had the paste() function that you removed (it worked but it was slower than your version tho so maybe yours can be fixed).

Also would it be possible to return the canvas coords of the layered image with it? This would be useful to generate a template directly with the layered image. (I was thinking of adding a [generate template] button or something after the layering)